### DISCUSSION TITLE
Warning when an interface with an optional function is declared

### DISCUSSION THREAD

**Discussion by wsaca** [vue-macros/vue-macros#234] [Q&A] [answered]
Before create an issue I would like to understand if the next warning could be resolved here:
```
[Vue warn]: Missing required prop: "onClick" 
  at <HelloWorld> 
  at <App>
```
I have this interface with two ways of declaring a function:
```ts
export interface HelloWorldProps {
  msg?: string;
  onClick?(): void;
  onFocus?: () => void;
}
```

Only If I use an arrow function it works.

Example:
https://stackblitz.com/edit/vitejs-vite-bcwnrm?file=src/components/types.ts

**Comment by sxzz** [comment, marked as answer]
Fixed, try the latest version.
